docs/2.9.0/docs/canton/usermanual/upgrading.rst

```diff
@@ -605,46 +683,78 @@ the upgrade test to understand the necessary downtime required for the migration
-we can just connect it to the new sync domain
+    There is no need for a one-step migration when upgrading from one release version to another,
+    as long as both release versions support the protocol version that your current sync domain uses.
+    Follow the steps for :ref:`upgrading the Canton binary <upgrade_canton_binary>` instead.
 
-.. snippet:: soft_migration_with_transfer
-    .. assert:: { participant.db.migrate(); true }
-    .. assert:: { participant.start(); true }
-    .. assert:: { participant.domains.connect_local(olddomain); true }
-    .. success:: participant.domains.connect_local(newdomain)
-    .. assert:: participant.domains.list_connected().map(_.domainAlias.unwrap).toSet == Set("newdomain", "olddomain")
+.. warning::
+
+    Every upgrade requires specific and thorough testing before applying it to a particular production
+    environment. Even though the one-step migration process has been tested through automatic and manual
+    tests, and its steps are known to work, additional measures and configuration may be required to
+    address the peculiarities of your environment.
+
+One-Step Migration Recipe for 2.9
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+General recipe to migrate from a sync domain running on a :ref:`2.3 <upgrade_to_2.3>`,
+:ref:`2.4 <upgrade_to_2.4>`, :ref:`2.5 <upgrade_to_2.5>`, :ref:`2.6 <upgrade_to_2.6>`,
+:ref:`2.7 <upgrade_to_2.7>` or :ref:`2.8 <upgrade_to_2.8>` release and protocol
+version 3 or 4 to a new sync domain running on the 2.9 release and protocol version 5:
+
+.. note::
+
+    We recommend using protocol version 5 even though 2.9 supports other protocol versions.
+
+#. Start a new sync domain running protocol version 5
+#. :ref:`Halt activity on the current sync domain <halt-activity-current-domain>`
+#. Wait for pending transactions to complete or time out
+#. Backup the current sync domain including participants
+#. Participants: Upgrade the binary to 2.9
+#. Participants: :ref:`Enable repair commands <enable-repair-commands>`
+#. All nodes: :ref:`Test the configuration <test-your-config>` considering these :ref:`details <test-your-config-details>`
+#. All nodes: :ref:`Apply the DB migrations <migrating_the_database>`
+#. Participants: Connect to the new sync domain, then disconnect from all domains
+#. Participants: Invoke :ref:`command<repair.migrate_domain>` ``repair.migrate_domain``
+#. Participants: Reconnect to the new sync domain
+#. Resume activity on the new sync domain
+
+.. _halt-activity-current-domain:
+
+Halt activity on the current sync domain
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+For a sync domain running on protocol version 3, set the resource limits to 0 on each participant.
+
+.. code-block:: text
+
+    participant.resources.set_resource_limits(ResourceLimits(Some(0), Some(0)))
+
+Otherwise, set the :ref:`dynamic domain parameter <dynamic_domain_parameters>` ``maxRatePerParticipant`` to 0.
```

Comment:
I would put this one first: "For a sync domain running protocol version 4 and above, ..."